beacon/src/beacon.rs

```diff
@@ -1,38 +1,38 @@
-use crate::{Entropy, Error, Result};
-use helium_proto::{
-    services::poc_lora,
-    {BlockchainRegionParamV1, DataRate},
-};
-use rand::{seq::SliceRandom, Rng, SeedableRng};
+use crate::{Entropy, Error, RegionParams, Result};
+use byteorder::{ByteOrder, LittleEndian};
+use helium_proto::{services::poc_lora, DataRate};
 use sha2::{Digest, Sha256};
-use std::time::{SystemTime, UNIX_EPOCH};
+use std::{
+    str::FromStr,
+    time::{SystemTime, UNIX_EPOCH},
+};
 
-pub const MAX_BEACON_V0_PAYLOAD_SIZE: usize = 10;
-pub const MIN_BEACON_V0_PAYLOAD_SIZE: usize = 5;
+pub const BEACON_PAYLOAD_SIZE: usize = 52;
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Beacon {
     pub data: Vec<u8>,
 
     pub frequency: u64,
     pub datarate: DataRate,
     pub remote_entropy: Entropy,
```

Comment:
as the verifier will have to reconstruct the beacon from submitted report data and then compare for equality, are the timestamps of the entropy going to cause issues ?  ie the beacon creation assigns timestamps of now() and none of those timestamps make it through to the report.

Thus when i convert from the received beacon_report to a beacon, i will likely have to assign current timestamps or use the timestamp from the report...but either way they wont match

Will PartialEq allow comparison of the timestamps to be skipped ?

Reply:
huh? no.. the `Beacon` itself has no notion of timestamps..only the reports. You can reconstruct a beacon from a report right?

Reply:
right but the beacon consists of a local and remote entropy field, each of which are of type Entropy which itself consists of a timestamp

Reply:
but only the entropy _data_ is used not the timestamp

Reply:
so maybe its all good then but my question is with PartialEq will those timestamps get in the way ?